src/fen_util.rs

```diff
@@ -91,6 +92,22 @@ pub fn valid_fen(fen: &str) -> bool {
             return false;
         }
     }
+
+    let castles = fen.split_whitespace().nth(2).unwrap();
+    if !unique_chars(castles) {
+        return false;
+    }
+    true
+}
+
+// TODO: Move this somewhere it makes more sense
+fn unique_chars(s: &str) -> bool {
+    let mut chars = std::collections::HashSet::new();
+    for c in s.chars() {
+        if !chars.insert(c) {
+            return false;
+        }
+    }
```

Comment:
I couldn't figure out how to say that `KK` was an invalid castle right. Maybe we should panic here instead of returning false.

Reply:
Maybe we can make an issue out of that. Panic if unable to parse.
But I'm worried about how that may constrict us in the future.

Reply:
I think returning false is fine for now. I also feel like it's time to make a util file; what do you think?

Reply:
I see a way to make this function a little shorter

Reply:
Yes let's make a util
- unique_chars
- Min
- Max
What else?